infra/app/build-repository/main.tf

```diff
@@ -9,6 +9,12 @@ locals {
     application_role = "build-repository"
     description      = "Backend resources required for storing built release candidate artifacts to be used for deploying to environments."
   })
+
+  # Get list of AWS account ids for the application environments that
+  # will need access to the build repository
+  app_account_names   = values(module.app_config.account_names_by_environment)
+  account_ids_by_name = data.external.account_ids_by_name.result
+  app_account_ids     = [for account_name in local.app_account_names : local.account_ids_by_name[account_name] if contains(keys(local.account_ids_by_name), account_name)]
```

Comment:
I don't totally follow this part, but I haven't used AWS in a multi-account scenario, so I might not be understanding why we need to gather account IDs in this way.

Reply:
yeah. if a project has a separate AWS account for each environment, then we need to set the IAM policy on the ECR to allow cross-account access, so we need a list of the accounts ids for this application. we have a map from environments to account names in the app-config, so we need to map the account names to account ids to get the list of account ids for all the environments for this application.

Reply:
Oh interesting. I had forgotten about the situation where you might want the ECR to be in one account, but then deploy from it to a complete environment (without ECR) in another. Got it.